### PR TITLE
CLOUDSTACK-8830 : VM snapshot creation fails for 12 min 

### PR DESCRIPTION
**Pull request by sarathkouk**
#ISSUE
VM snapshot fails for 12 min after instance creation

#FIX 
This PR adds a marvin test case for the fix at #1677
Test case add to check whether VMSnapshot is successful from the moment the VM is ready

### CODE REVIEW

**Comment by sarathkouk** [comment]
Travis CI build failed due to smoke/test_outofbandmanagement which is not connected to this code.

File: test/integration/component/test_escalations_vmware.py [hunk 1]
```diff
@@ -331,3 +334,46 @@ def test3_attach_ISO_in_RHEL7OSVM(self):
...
+            "Snapshot creation failed"
+        )
+
+        time_elapsed = time.time() - start_time
+
+        self.assertTrue(
+            time_elapsed < 600,
+            "Snapshot did not get created in the first 10 minutes after VM creation"
+        )
+
```
Comment (sureshanaparti) [inline test/integration/component/test_escalations_vmware.py]: Sometimes VM snapshot may take more than 10 mins. In that case this assertion would be false.

File: test/integration/component/test_escalations_vmware.py [hunk 2]
```diff
@@ -331,3 +334,46 @@ def test3_attach_ISO_in_RHEL7OSVM(self):
...
+        snapshot_created_1 = VmSnapshot.create(
+            self.userapiclient,
+            vm.id
+        )
+        self.assertIsNotNone(
+            snapshot_created_1,
+            "Snapshot creation failed"
+        )
+
```
Comment (sureshanaparti) [inline test/integration/component/test_escalations_vmware.py]: If create VM snapshot fails immediately after instance creation, the issue might exists.

**Comment by sureshanaparti** [comment]
Test code LGTM

**Comment by Divyanimmagadda** [comment]
LGTM

**Comment by karuturi** [comment]
Can you post test results?

**Pull request merged**